zml/aio/torch.zig

```diff
@@ -312,102 +312,97 @@ pub const PickleData = struct {
                 try self.parseValue(allocator, store, prefix, build.args);
             },
             .pers_id => |pers_id| try self.parseValue(allocator, store, prefix, pers_id.ref),
-            .seq => |*seq| switch (seq[0]) {
-                .list, .tuple, .set, .frozen_set => {
-                    const elemCheck = struct {
-                        fn call(comptime T: ValueType) fn (v: Value) bool {
-                            return struct {
-                                fn call(val: Value) bool {
-                                    return val == T;
+            .seq => |seq| {
+                switch (seq.type) {
+                    .list, .tuple, .set, .frozen_set => {
+                        if (seq.values.len == 0) return;
+                        var valid_slice = true;
+                        switch (seq.values[0]) {
+                            inline .int64, .float64, .boolval => |val0, tag| {
+                                const ItemType = switch (tag) {
+                                    .int64 => i64,
+                                    .float64 => f64,
+                                    .boolval => bool,
+                                    else => unreachable,
+                                };
+                                var values: std.ArrayListUnmanaged(ItemType) = .{};
```

Comment:
can you use ensureCapacityPrecise on values ? that way `toOwnedSlice` isn't needed anymore